src/gt4py/next/program_processors/runners/dace_fieldview/gtir_to_sdfg.py

```diff
@@ -0,0 +1,248 @@
+            head_state = sdfg.add_state_after(entry_state, "init_temps", assignments=temp_symbols)
+        else:
+            head_state = entry_state
+
+        # add non-transient arrays and/or SDFG symbols for the program arguments
+        for param in node.params:
+            assert isinstance(param.type, ts.DataType)
+            self._add_storage(sdfg, str(param.id), param.type)
+
+        # visit one statement at a time and expand the SDFG from the current head state
+        for i, stmt in enumerate(node.body):
+            # include `debuginfo` only for `ir.Program` and `ir.Stmt` nodes: finer granularity would be too messy
+            head_state = sdfg.add_state_after(head_state, f"stmt_{i}")
+            head_state._debuginfo = dace_fieldview_util.debuginfo(stmt, sdfg.debuginfo)
+            self.visit(stmt, sdfg=sdfg, state=head_state)
+
+        sdfg.validate()
+        return sdfg
+
+    def visit_SetAt(self, stmt: itir.SetAt, sdfg: dace.SDFG, state: dace.SDFGState) -> None:
+        """Visits a `SetAt` statement expression and writes the local result to some external storage.
+
+        Each statement expression results in some sort of dataflow gragh writing to temporary storage.
+        The translation of `SetAt` ensures that the result is written back to the target external storage.
+        """
+
+        expr_nodes = self._visit_expression(stmt.expr, sdfg, state)
+
+        # the target expression could be a `SymRef` to an output node or a `make_tuple` expression
+        # in case the statement returns more than one field
+        target_nodes = self._visit_expression(stmt.target, sdfg, state)
+
+        # convert domain expression to dictionary to ease access to dimension boundaries
+        domain = dace_fieldview_util.get_domain_ranges(stmt.domain)
+
+        for expr_node, target_node in zip(expr_nodes, target_nodes, strict=True):
+            target_array = sdfg.arrays[target_node.data]
+            assert not target_array.transient
+            target_symbol_type = self.symbol_types[target_node.data]
+
+            if isinstance(target_symbol_type, ts.FieldType):
+                subset = ",".join(
+                    f"{domain[dim][0]}:{domain[dim][1]}" for dim in target_symbol_type.dims
+                )
+            else:
+                assert len(domain) == 0
+                subset = "0"
+
+            state.add_nedge(
+                expr_node,
+                target_node,
+                dace.Memlet(data=target_node.data, subset=subset),
+            )
+
+    def visit_FunCall(
+        self, node: itir.FunCall, sdfg: dace.SDFG, head_state: dace.SDFGState
+    ) -> list[gtir_builtin_translators.SDFGField]:
+        # use specialized dataflow builder classes for each builtin function
+        if cpm.is_call_to(node.fun, "as_fieldop"):
+            return gtir_builtin_translators.AsFieldOp(node.args)(node.fun, sdfg, head_state, self)
```

Comment:
As mentioned above, I think this should just be:
```suggestion
            return gtir_builtin_translators.AsFieldOp(node, sdfg, head_state, self)
```

Reply:
I cannot call the primitive translator in the way you suggest. Currently I have to do in this way:
`gtir_builtin_translators.AsFieldOp()(node, sdfg, head_state, self)`

I have the impression that this pattern is not really good. What I need is basically a static class method, or a module function, for each primitive.